packages/vite-plugin-svelte-docgen/src/transform.js

```diff
@@ -63,6 +48,26 @@ export function transform_encoded(ast) {
 						},
 					});
 				}
+				const is_toplevel =
+					/** @type {AST.ObjectExpression} */ (ctx.path[ctx.path.length - 1]).properties.find(
+						(p) => p.type === "Property" && p.key.type === "Literal" && p.key.value === "props",
+					) !== undefined;
+				if (is_toplevel) {
+					/** Revive those keys values as {@link Map} */
+					if (["events", "exports", "props", "types", "slots"].includes(node.key.value)) {
```

Comment:
Since types can also appear within Type, handle these only at the top-level object.